apis/object/v1alpha1/types.go

```diff
@@ -123,6 +122,23 @@ type ObjectSpec struct {
 	// +kubebuilder:default=Default
 	ManagementPolicy `json:"managementPolicy,omitempty"`
 	References       []Reference `json:"references,omitempty"`
+	Readiness        *Readiness  `json:"readiness,omitempty"`
+}
+
+// Readiness defines how the object's readiness condition should be computed,
+// if not specified it will be considered ready as soon as the underlying external
+// resource is considered up-to-date.
+type Readiness struct {
+	// FromObject configures whether the Object's readiness should be derived from
+	// the readiness of the underlying external resource.
+	// +optional
+	FromObject *bool `json:"fromObject,omitempty"`
```

Comment:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Not a hard and fast rule, but it's generally a good idea to avoid boolean fields.

> Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. Plan for future expansions by describing the policy options explicitly as a string type alias (e.g. TerminationMessagePolicy).

I could see this being another enum like `readiness.policy` or `readiness.source` with two options (for now):

1. Successful creation of object (I think this is the default today, right? I didn't check).
1. Derive from readiness of object. (Would this also need optional configuration - i.e. what indicates the object is ready?)

Reply:
Our idea was to leave it to a simple Boolean now, but put it in a dedicated stanza that would allow us to add a list of readinessChecks if required down the line:

```yaml
readiness:
  fromObject: false
  checks:
  - type: MatchCondition
    matchCondition: ...
```

But we could go with:

```yaml
readiness:
  type: FromChecks #FromObject
  checks:
  - type: MatchCondition
    matchCondition: ...
```
Which would mean now we only implement:

```yaml
readiness:
  type: FromObject
```

Reply:
Throwing out another possibility.

```
readiness:
  policy: SuccessfullCreate (default) # DeriveFromObject, PassingChecks
```

Reply:
which would be self-documenting w.r.t. the current behaviour too, love it!

Reply:
implemented and updated the PR description accordingly, kept it in a separate commit, but I'd squash everything before merging.